### ISSUE TITLE
option return, gives  C error. This should never happen...

### ISSUE THREAD

**Issue by despiegk** [vlang/v#7303]
result:

```
bash-3.2$ ./v run test.v
...
==================
                       ^
/var/folders/zk/gpydw9_j3qv2t0hmbcvts7z40000gn/T/v/test.7543210048261177675.tmp.c:11340:8: error: use of undeclared identifier '_t260'
                if (!_t260.ok) {
                     ^
/var/folders/zk/gpydw9_j3qv2t0hmbcvts7z40000gn/T/v/test.7543210048261177675.tmp.c:11341:17: error: use of undeclared identifier '_t260'
                        string err = _t260.v_error;
                                     ^
/var/folders/zk/gpydw9_j3qv2t0hmbcvts7z40000gn/T/v/test.7543210048261177675.tmp.c:11342:18: error: use of undeclared identifier '_t260'
                        int errcode = _t260.ecode;
                                      ^
/var/folders/zk/gpydw9_j3qv2t0hmbcvts7z40000gn/T/v/test.7543210048261177675.tmp.c:11352:11: error: unexpected type name 'Option_finder__Page': expected expression
                        return Option_finder__Page _t264 = finder__Site_page_get(&site, name_lower);
...
==================
(Use `v -cg` to print the entire error message)
```

code used:

- https://github.com/crystaluniverse/vlang_tools/blob/main/vlang_tools/test.v

to reproduce, checkout the git repo and go inside the dir ...vlang_tools/blob/main/vlang_tools

run the run.sh

ofocurse it could be I made a beginners mistake but then I guess error should be different, it that's the case

**Comment by despiegk** [comment]
i managed to fix it but still error is not good I think

![image](https://user-images.githubusercontent.com/6021844/102008551-cd2e7d80-3d31-11eb-9e0a-c816b0fcb852.png)

I had to put the page on line and then only return, the return was on same line

**Comment by medvednikov** [comment]
Code resulting in the error:

```
return site.page_get(name_lower) or {return error("")}
```

@joe-conigliaro how should we handle such expressions?

**Comment by despiegk** [comment]
we can close there, but it would be great if the compiler can catch these issues and report them to the developer

**Comment by medvednikov** [comment]
This should stay open until it's fixed.

**Comment by spaceface777** [comment]
Fixed in #9021.
